schematic_api/api/routes.py

```diff
@@ -206,130 +220,193 @@ def get_temp_jsonld(schema_url):
     # get path to temporary JSON-LD file
     return tmp_file.name
 
-# @before_request
-def get_manifest_route(schema_url: str, use_annotations: bool, dataset_ids=None, asset_view = None, output_format=None, title=None, access_token=None):
-    """Get the immediate dependencies that are related to a given source node.
-        Args:
-            schema_url: link to data model in json ld format
-            title: title of a given manifest. 
-            dataset_id: Synapse ID of the "dataset" entity on Synapse (for a given center/project).
-            output_format: contains three option: "excel", "google_sheet", and "dataframe". if set to "excel", return an excel spreadsheet
-            use_annotations: Whether to use existing annotations during manifest generation
-            asset_view: ID of view listing all project data assets. For example, for Synapse this would be the Synapse ID of the fileview listing all data assets for a given project.
-            access_token: Token
-        Returns:
-            Googlesheet URL (if sheet_url is True), or pandas dataframe (if sheet_url is False).
-    """
-
-    # call config_handler()
-    config_handler(asset_view = asset_view)
+@dataclass
+class ManifestGeneration():
+    schema_url: str
+    data_type: List[str]
+    asset_view: Optional[str] = None
+    dataset_id: Optional[List[str]] = None
+    title: Optional[str] = None
+    output_format: Optional[str] = field(default_factory=lambda: "google_sheet")
+    use_annotations: Optional[bool] = field(default_factory=bool)
+    
+    def _load_config_(self, app:Flask=app) -> CONFIG:
+        """load configuration file and update asset view if needed
 
-    # get path to temporary JSON-LD file
-    jsonld = get_temp_jsonld(schema_url)
+        Args:
+            app (Flask, optional): Flask app object. Defaults to current flask object.
 
-    # Gather all data_types to make manifests for.
-    all_args = connexion.request.args
-    args_dict = dict(all_args.lists())
-    data_type = args_dict['data_type']
-    
-    # Gather all dataset_ids
-    try:
-        dataset_ids = args_dict['dataset_id']
-    except:
-        pass
-    
-    if dataset_ids:
-        # Check that the number of submitted data_types matches
-        # the number of dataset_ids (if applicable)
-        len_data_types = len(data_type)
-        len_dataset_ids = len(dataset_ids)
-        
-        try:
-            len_data_types == len_dataset_ids
-        except:
+        Returns:
+            CONFIG: schematic configuration object
+        """
+        return config_handler(app=app, asset_view=self.asset_view)
+
+    def __post_init__(self):
+        # Note: all the code below could be done better using "@validate" in python3.10. 
```

Comment:
Yes. That makes sense. I think I tried implementing the validator but I thought it would only work for BaseModel. Let me change that part to use @validator. Thanks for the feedback Bruno